Agents/prosumer_model.py

```diff
@@ -3,12 +3,34 @@
 from abc import ABC, abstractmethod
 
 
-def average(self, dates):
+def average(self:list, dates:list) -> list:
+    """
+    Calculate the average of the hourly curve for the day
+
+    ...
+
+    Parameters
+    ----------
+    self : list
+        A 2D list containing the calculated energy values for a specific date
+    dates : list
+        A list of dates for a specific date in "YYYY-MM-DD hh:mm:ss" format
+
+    Returns
+    -------
+    hourly_average : list
+        A 2D list containin the calculated hourly averages for a given date range
```

Comment:
Thanks for the clarification. If hourly_average is not typed as an array, we can leave it as a list. However, it might be better to call it a nested list in the description.